control-plane/agents/src/bin/ha/node/server.rs

```diff
@@ -100,11 +113,11 @@ fn disconnect_controller(ctrlr: &NvmeController) -> Result<(), SvcError> {
 impl NodeAgentSvc {
     /// Translate new path URI into connection arguments for Subsystem connect API.
     fn get_nvmf_connection_args(&self, new_path: &str) -> Option<ConnectArgs> {
-        let uri = new_path.parse::<Uri>().ok()?;
+        let parsed_path = parse_uri(new_path)?;
```

Comment:
Any reason for changing these? The previous one seemed cleaner.

Reply:
The reason for changing it is, since now its being used at multiple places, I wanted to map and return error from one place rather than from each getter, but yeah using a tuple is indeed not cleaner. I will use a `ParsedUri` type for the same.